agent/api/container.go

```diff
@@ -548,3 +553,38 @@ func (c *Container) GetHealthStatus() HealthStatus {
 
 	return copyHealth
 }
+
+// UpdateTransitioningStatusUnsafe updates the container transitioning status
+func (c *Container) UpdateTransitioningStatusUnsafe(knownStatus ContainerStatus) {
+	if c.transitioningStatus == ContainerNotTransition {
+		return
+	}
+
+	// Check if the container transition has already finished
+	if containerTransitioningStatusMapping[c.transitioningStatus] <= knownStatus {
+		c.transitioningStatus = ContainerNotTransition
```

Comment:
do we need a lock for this? also should we just be using the setter here?

Reply:
This method should be called within SetKnownStatus, which has already held the lock. Setter is holding a lock where we don't need a lock here, that's why the name xxxUnsafe.